xrpl/asyncio/clients/json_rpc_base.py

```diff
@@ -36,4 +37,9 @@ async def request_impl(self: JsonRpcBase, request: Request) -> Response:
                 self.url,
                 json=request_to_json_rpc(request),
             )
+            if response.status_code != 200:
```

Comment:
Should we allow any `200` response code? - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses

Reply:
Maybe 204 too.  That is very common.  It could be with the RPC api only 200s are sent.

Reply:
That might be a question for the rippled team

Reply:
From Elliot's suggestion, I changed this to raise `XRPLRequestFailureException` when the response can't be JSON decoded.